bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.juicenet.internal.api;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDevice;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiDeviceStatus;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiInfo;
+import org.openhab.binding.juicenet.internal.api.dto.JuiceNetApiTouSchedule;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link JuiceNetApi} is responsible for implementing the api interface to the JuiceNet cloud server
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetApi {
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetApi.class);
+
+    private static final String API_HOST = "https://jbv1-api.emotorwerks.com/";
+    private static final String API_ACCOUNT = API_HOST + "box_pin";
+    private static final String API_DEVICE = API_HOST + "box_api_secure";
+
+    protected String apiToken = "";
+    protected HttpClient httpClient;
+    protected ThingUID bridgeUID;
+
+    public enum ApiCommand {
+        GET_ACCOUNT_UNITS("get_account_units", API_ACCOUNT),
+        GET_STATE("get_state", API_DEVICE),
+        SET_CHARGING_LIMIT("set_limit", API_DEVICE),
+        GET_SCHEDULE("get_schedule", API_DEVICE),
+        SET_SCHEDULE("set_schedule", API_DEVICE),
+        GET_INFO("get_info", API_DEVICE),
+        SET_OVERRIDE("set_override", API_DEVICE);
+
+        final String command;
+        final String uri;
+
+        ApiCommand(String command, String uri) {
+            this.command = command;
+            this.uri = uri;
+        }
+    }
+
+    public JuiceNetApi(HttpClient httpClient, ThingUID bridgeUID) {
+        this.bridgeUID = bridgeUID;
+        this.httpClient = httpClient;
+    }
+
+    public void initialize(String apiToken) {
```

Comment:
Consider renaming method now that it does only one thing:
```suggestion
    public void setApiToken(String apiToken) {
```

This would be more straight-forward and readable.